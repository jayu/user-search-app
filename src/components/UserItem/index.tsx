import React from 'react';
import styles from './styles.module.css';

type Props = {
  name: string;
  username: string;
};

const UserItem = ({ name, username }: Props) => {
  return (
    <p className={styles.userContainer}>
      <span>{name}</span>
      <span className={styles.userName}>@{username}</span>
    </p>
  );
};

export default UserItem;
