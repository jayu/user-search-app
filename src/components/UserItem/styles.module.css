.userContainer {
  font-size: 1rem;
}

.userName {
  color: #7a7a7a;
  margin-left: 15px;
}
