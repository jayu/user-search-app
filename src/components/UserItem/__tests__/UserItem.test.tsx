import React from 'react';
import { render } from '@testing-library/react';
import snapshotDiff from 'snapshot-diff';
import UserItem from '../index';

test('It should render User information', () => {
  const { getByText } = render(<UserItem name="Jakub" username="jayu" />);

  const name = getByText(/Jakub/);
  const username = getByText(/@jayu/);

  expect(name).toBeInTheDocument();
  expect(username).toBeInTheDocument();
});

test('It should assert difference between two UserItems', () => {
  expect(
    snapshotDiff(<UserItem name="Jakub" username="jayu" />, <UserItem name="Mathew" username="math" />),
  ).toMatchSnapshot();
});
