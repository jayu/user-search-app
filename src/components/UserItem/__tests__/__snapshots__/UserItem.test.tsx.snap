// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should assert difference between two UserItems 1`] = `
"Snapshot Diff:
- <UserItem name=\\"Jakub\\" username=\\"jayu\\" />
+ <UserItem name=\\"Mathew\\" username=\\"math\\" />

  <p
    className=\\"userContainer\\"
  >
    <span>
-     Jakub
+     Mathew
    </span>
    <span
      className=\\"userName\\"
    >
      @
-     jayu
+     math
    </span>
  </p>"
`;
